addons/mrp_account/models/mrp_production.py

```diff
@@ -48,27 +55,40 @@ def _prepare_wc_analytic_line(self, wc_line):
         value = hours * wc.costs_hour
         account = wc.costs_hour_account_id.id
         return {
-            'name': wc_line.name + ' (H)',
+            'name': _("[WC] %(name)s", name=wc_line.name),
             'amount': -value,
             'account_id': account,
             'ref': wc.code,
             'unit_amount': hours,
             'company_id': self.company_id.id,
+            'category': 'manufacturing_order',
         }
 
     def _costs_generate(self):
         """ Calculates total costs at the end of the production.
         """
         self.ensure_one()
+        # we use SUPERUSER_ID as we do not guarantee an mrp user
+        # has access to account analytic lines but still should be
+        # able to produce orders
         AccountAnalyticLine = self.env['account.analytic.line'].sudo()
-        for wc_line in self.workorder_ids.filtered('workcenter_id.costs_hour_account_id'):
-            vals = self._prepare_wc_analytic_line(wc_line)
-            precision_rounding = (wc_line.workcenter_id.costs_hour_account_id.currency_id or self.company_id.currency_id).rounding
-            if not float_is_zero(vals.get('amount', 0.0), precision_rounding=precision_rounding):
-                # we use SUPERUSER_ID as we do not guarantee an mrp user
-                # has access to account analytic lines but still should be
-                # able to produce orders
-                AccountAnalyticLine.create(vals)
+        (self.workorder_ids.mo_analytic_account_line_id | self.workorder_ids.wc_analytic_account_line_id).unlink()
+        vals_list = []
+        for wo in self.workorder_ids:
+            hours = wo.duration / 60.0
+            value = -hours * wo.workcenter_id.costs_hour
+            mo_account = wo.production_id.analytic_account_id
+            wc_account = wo.workcenter_id.costs_hour_account_id
+            # for mo analytic account
+            precision_rounding = (mo_account.currency_id or self.company_id.currency_id).rounding
+            if mo_account and not float_is_zero(value, precision_rounding=precision_rounding):
+                vals_list.append(wo._prepare_analytic_line(mo_account, hours, value))
+            # for work center analytic account
+            precision_rounding = (wc_account.currency_id or self.company_id.currency_id).rounding
+            if (wc_account and wc_account != mo_account and
+                not float_is_zero(value, precision_rounding=precision_rounding)):
+                vals_list.append(wo._prepare_analytic_line(wc_account, hours, value))
+        AccountAnalyticLine.create(vals_list)
```

Comment:
self.env['account.analytic.line'].sudo() and remove AccountAnalyticLine variable